### DISCUSSION TITLE
Stake out line and polygons

### DISCUSSION THREAD

**Discussion by 4SS3** [opengisch/QField#5848] [Ideas]
I am missing the function for staking out lines and polygons.
I often find myself in the situation where I need to know the distance to a line or a polygon and also the distance to the start and end point of a line or a polygon edge. 
Is this function anyone else sees the need for, or is it just me?

**Comment by nirvn** [comment]
@4SS3 , you can set any geometry ( line or polygon ) from vector layers within your project as a destination feature; when you do that, you will be allow to cycle through the geometry's centroid _as well as_ every single vertex, which would give you the start and end point of a line.

**Reply by 4SS3** [reply]
Yes, but I also need to know the perpendicular distance to lines and polygon edges.
With this function in qfield I can ditch the separate surveying app and get a nice workflow with qgis and qfield cloud.

**Reply by nirvn** [reply]
Do you have an image to represent what "perpendicular distance to lines and polygon edges" means?

**Reply by VxTedxV** [reply]
I think 4SS3 means the shortest distance between two entities (which is perpendicular to one of them).

**Comment by 4SS3** [comment]
The position is the red dot.
I want to know the distance to the line [**S**] 
I also want to know how far from the stating point of the line that my position is [**D**]

The calculations of the distances to the line should also extend beyond the starting and end points with an indication that the position is outside the boundary points of the line.

It is desirable that the stake out function should have support for polylines and polygons.

![image](https://github.com/user-attachments/assets/890c1446-0a1f-4017-91a4-b2a85a8596ac)

**Reply by tornisteri** [reply]
Did you find out whether this is possible? This is exactly what I'm trying to figure out as well.

**Reply by 4SS3** [reply]
Yes, this is doable. It's not even that hard to do.
There is even existing functions in qgis to do the calculations.

This function is in almost every surveying app. This is also what stops us from switching to qfield for our field work.
